src/common/reducers/modals/reducers.js

```diff
@@ -8,15 +8,21 @@ export default function modals(state = [], action) {
         modalProps: action.modalProps
       });
     case ActionTypes.UNMOUNTING_MODAL:
-      return [
-        ...state.slice(0, state.length - 1),
-        {
-          ...state[state.length - 1],
-          unmounting: true
-        }
-      ];
+      if (state.length > 0) {
+        return [
+          ...state.slice(0, state.length - 1),
+          {
+            ...state[state.length - 1],
+            unmounting: true
+          }
+        ];
+      }
+      break;
     case ActionTypes.CLOSE_MODAL:
-      return state.slice(0, state.length - 1);
+      if (state.length > 0) {
+        return state.slice(0, state.length - 1);
+      }
+      break;
```

Comment:
There are a few issues with this approach.
Firstly, we always want to return a new state from the reducer. Returning undefined will result in the renderer thread crashing. Also, there's no need to `break` since we would already return the state